client/src/plugins/zeebe-plugin/start-instance-plugin/StartInstancePlugin.js

```diff
@@ -48,84 +50,294 @@ export default class StartInstancePlugin extends PureComponent {
 
-    this.props.displayNotification({
-      type: 'success',
-      title: 'Process instance started',
-      content: content,
-      duration: 8000
-    });
+      // (3.1.1) handle start instance error
+      if (!startInstanceResult.success) {
+        this.handleStartError(startInstanceResult.response);
+        return;
+      }
+
+      // (3.1.2) handle start instance success
+      this.handleStartSuccess(startInstanceResult, endpoint);
+
+    } catch (error) {
+
+      // (3.1.3) handle start instance exception
+      this.handleStartError(error);
+    }
   }
 
-  onOverlayClose = async () => {
-    this.props.broadcastMessage('cancel');
-    this.setState({ activeButton: false });
+  decorateVariables = (startConfiguration) => {
+    let {
+      variables
+    } = startConfiguration;
+
+    if (variables && variables.trim().length > 0) {
+      variables = JSON.parse(variables);
+    } else {
+      variables = null;
+    }
+
+    return { ...startConfiguration, variables };
   }
 
-  deployActiveTab() {
+  async getDeployConfig() {
     const deferred = pDefer();
     const body = {
       isStart: true,
       skipNotificationOnSuccess: true,
       done: deferred.resolve,
       anchorRef: this._anchorRef,
-      onClose: () => { this.setState({ activeButton: false }); }
+      notifyResult: true,
+      onClose: () => {
+        this.setState({
+          activeButton: false,
+          overlayState: null
+        });
+      }
     };
 
-    this.setState({ activeButton: true });
+    this.props.broadcastMessage('getDeployConfig', body);
 
-    this.props.broadcastMessage('deploy', body);
+    return deferred.promise;
+  }
+
+  async deployWithConfig(deploymentConfig) {
+    const deferred = pDefer();
+    const body = {
+      isStart: true,
+      skipNotificationOnSuccess: true,
+      done: deferred.resolve,
+      anchorRef: this._anchorRef,
+      tab: this.state.activeTab,
+      deploymentConfig
+    };
+
+    this.props.broadcastMessage('deployWithConfig', body);
 
     return deferred.promise;
   }
 
+  handleStartSuccess(processInstance, endpoint) {
+    const {
+      displayNotification
+    } = this.props;
+
+    const content = endpoint.targetType === CAMUNDA_CLOUD ?
+      <CloudLink endpoint={ endpoint } response={ processInstance.response } />
+      : null;
+
+    displayNotification({
+      type: 'success',
+      title: 'Process instance started',
+      content: content,
+      duration: 8000
+    });
+  }
+
+  handleStartError(error) {
+    const {
+      log,
+      displayNotification
+    } = this.props;
+
+    const logMessage = {
+      category: 'start-instance-error',
+      message: error.details || error.message
+    };
+
+    const content = <button
+      onClick={ ()=> log(logMessage) }>
+      See the log for further details.
+    </button>;
+
+    displayNotification({
+      type: 'error',
+      title: 'Starting process instance failed',
+      content: content,
+      duration: 4000
```

Comment:
Just curious: why do we display success longer than failures (8s vs 4s)? Is that a best practice for notifications?

Reply:
I'm not sure why this is. I remember we had many different durations and when I worked on the notifications I changed them to be more consistent based on the original value.

It could be that error notifications didn't use to have any actions and just a "Open to see log" plain text, therefore not requiring any user action. But we changed this to also be a link to open the log so maybe we can revisit the durations. What are your thoughts on this @andreasgeier ?